### PR TITLE
BUG: Fix for xlabel/ylabel in barh plot

### LINKED ISSUE #45144: BUG: xlabel/ylabel don't work as expected with barh plots
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame(
    {'A': [4, 2, 1, 2, 5, 2]}, index=pd.RangeIndex(6, name='B')
)

df.plot(kind='barh', xlabel='x', ylabel='y')
```

### Issue Description

With `bar` plots, we can pass `xlabel` and `ylabel` to `plot()` and have those values set as the x- and y-axis labels (overriding any defaults). However, if you try this with a `barh` plot both `xlabel` and `ylabel` update the y-axis label, and the only way to update the x-axis label is after the fact with `ax.set_xlabel()`. Here's the result for the code above:

image

### Expected Behavior

With the code provided above, I would expect this:

image

In addition, providing only `xlabel` should label the x-axis only (currently, this updates the y-axis label, which is confusing).

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 945c9ed766a61c7d2c0a7cbb251b6edebf9cb7d5
python           : 3.9.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.6.0
Version          : Darwin Kernel Version 19.6.0: Thu Sep 16 20:58:47 PDT 2021; root:xnu-6153.141.40.1~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.3.4
numpy            : 1.20.3
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 21.2.4
setuptools       : 52.0.0.post20210125
Cython           : 0.29.25
pytest           : 6.2.5
hypothesis       : 6.29.3
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.2
IPython          : 7.29.0
pandas_datareader: 0.10.0
bs4              : None
bottleneck       : 1.3.2
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.2
numexpr          : 2.7.3
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by stefmolin**
- [X] closes #45144
- [X] tests added / passed
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [X] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.4.0.rst [hunk 1]
```diff
@@ -757,6 +757,8 @@ Timedelta
 ^^^^^^^^^
 - Bug in division of all-``NaT`` :class:`TimeDeltaIndex`, :class:`Series` or :class:`DataFrame` column with object-dtype arraylike of numbers failing to infer the result as timedelta64-dtype (:issue:`39750`)
 - Bug in floor division of ``timedelta64[ns]`` data with a scalar returning garbage values (:issue:`44466`)
+- Bug in :class:`Timedelta` now properly taking into account any nanoseconds contribution of any kwarg (:issue:`43764`)
```
Comment (phofl) [inline doc/source/whatsnew/v1.4.0.rst:760]: This is form somewhere else?

File: doc/source/whatsnew/v1.4.0.rst [hunk 2]
```diff
@@ -757,6 +757,8 @@ Timedelta
...
+-
```
Comment (attack68) [inline doc/source/whatsnew/v1.4.0.rst:761]: these extra edits in the whatsnew seem to have crept in? they seem unrelated to your PR?

File: doc/source/whatsnew/v1.4.0.rst (hunk 1 above)
Comment (stefmolin) [reply doc/source/whatsnew/v1.4.0.rst]: It looks like something was merged between my writing up the PR and submitting it. I just did a fetch and merge on my branch.

File: doc/source/whatsnew/v1.4.0.rst (hunk 2 above)
Comment (stefmolin) [reply doc/source/whatsnew/v1.4.0.rst]: It looks like something was merged between my writing up the PR and submitting it. I just did a fetch and merge on my branch.

File: doc/source/whatsnew/v1.4.0.rst [hunk 3]
```diff
@@ -884,6 +884,7 @@ Period
 Plotting
 ^^^^^^^^
 - When given non-numeric data, :meth:`DataFrame.boxplot` now raises a ``ValueError`` rather than a cryptic ``KeyError`` or ``ZeroDivisionError``, in line with other plotting functions like :meth:`DataFrame.hist`. (:issue:`43480`)
+- Bug in :meth:`DataFrame.plot.barh` that prevented labeling the x-axis and ``xlabel`` updating the y-axis label (:issue:`45144`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:887]: can you move to 1.5

File: pandas/plotting/_matplotlib/core.py [hunk 4]
```diff
@@ -767,8 +767,10 @@ def _get_index_name(self) -> str | None:
                 name = pprint_thing(name)
 
         # GH 9093, override the default xlabel if xlabel is provided.
-        if self.xlabel is not None:
+        if self.xlabel is not None and self.orientation == "vertical":
             name = pprint_thing(self.xlabel)
+        if self.ylabel is not None and self.orientation == "horizontal":
```
Comment (jreback) [inline pandas/plotting/_matplotlib/core.py:772]: so if these are provided and there is an incorrect orientation it is ignored? maybe could add a warning for that case

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.4.0.rst (hunk 3 above)
Comment (stefmolin) [reply doc/source/whatsnew/v1.4.0.rst]: Sure &ndash; moved to 1.5

File: pandas/plotting/_matplotlib/core.py (hunk 4 above)
Comment (stefmolin) [reply pandas/plotting/_matplotlib/core.py]: Done &ndash; added a warning to both cases.

Comment (stefmolin) [reply pandas/plotting/_matplotlib/core.py]: Actually, I don't think it makes sense to have the warning here. Before this PR, running the below would have caused the `xlabel` argument to be used for the y-axis:

```python
import pandas as pd

df = pd.DataFrame(
    {'A': [4, 2, 1, 2, 5, 2]}, index=pd.RangeIndex(6, name='B')
)
df.plot(kind='barh', xlabel='x')
```

A warning for this case with the new code doesn't seem correct because now by "ignoring" the `xlabel`/`ylabel` values at this point in the code, we are just ensuring that when the axis labels are applied later it is done correctly. For example, with the new code we get a warning, but the output is exactly what was intended:
Screen Shot 2022-01-10 at 6 02 31 PM

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @stefmolin very nice!
